integration/init_test.go

```diff
@@ -56,21 +59,55 @@ func TestInit(t *testing.T) {
 				"-a", `{"builder":"Docker","payload":{"path":"leeroy-web/Dockerfile"},"image":"gcr.io/k8s-skaffold/leeroy-web"}`,
 			},
 		},
+	}
+	for _, test := range tests {
+		testutil.Run(t, test.name, func(t *testutil.T) {
+			initArgs := append([]string{"--force"}, test.args...)
+			skaffold.Init(initArgs...).InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t.T)
+
+			checkGeneratedConfig(t, test.dir)
+
+			skaffold.Diagnose().InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t.T)
+		})
+	}
+}
+
+func TestInitCompose(t *testing.T) {
+	if testing.Short() || RunOnGCP() {
+		t.Skip("skipping kind integration test")
+	}
+
+	tests := []struct {
+		name string
+		dir  string
+		args []string
+	}{
 		{
 			name: "compose",
 			dir:  "testdata/init/compose",
 			args: []string{"--compose-file", "docker-compose.yaml"},
 		},
 	}
 	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			ns, _, deleteNs := SetupNamespace(t)
+		testutil.Run(t, test.name, func(t *testutil.T) {
+			ns, _, deleteNs := SetupNamespace(t.T)
 			defer deleteNs()
 
 			initArgs := append([]string{"--force"}, test.args...)
-			skaffold.Init(initArgs...).InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t)
+			skaffold.Init(initArgs...).InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t.T)
+
+			checkGeneratedConfig(t, test.dir)
 
-			skaffold.Run().InDir(test.dir).WithConfig("skaffold.yaml.out").InNs(ns.Name).RunOrFail(t)
+			skaffold.Run().InDir(test.dir).WithConfig("skaffold.yaml.out").InNs(ns.Name).RunOrFail(t.T)
 		})
 	}
 }
+
+func checkGeneratedConfig(t *testutil.T, dir string) {
+	expectedOutput, err := ioutil.ReadFile(filepath.Join(dir, "skaffold.yaml"))
+	t.CheckNoError(err)
+
+	output, err := ioutil.ReadFile(filepath.Join(dir, "skaffold.yaml.out"))
+	t.CheckNoError(err)
+	t.CheckDeepEqual(string(expectedOutput), string(output))
```

Comment:
This might become a bit brittle over time, we could add some guidance to make maintenance a bit less painful in the comment here or as part of the error message that: "if you see failure here, inspect the changes. In case the changes in config are expected (e.g. version change), run `find integration -name skaffold.yaml.out -execdir mv skaffold.yaml.out skaffold.yaml \;` to override all the skaffold.yamls to the latest version.

Reply:
on the long run we might want to make this a bit smarter to ignore guaranteed cases of string changes that are okay - e.g. version change.

Reply:
The version change should be made automatically by our script